skbio/diversity/_driver.py

```diff
@@ -0,0 +1,305 @@
+        'enspie': skbio.diversity.alpha.enspie,
+        'esty_ci': skbio.diversity.alpha.esty_ci,
+        'faith_pd': skbio.diversity.alpha.faith_pd,
+        'fisher_alpha': skbio.diversity.alpha.fisher_alpha,
+        'goods_coverage': skbio.diversity.alpha.goods_coverage,
+        'heip_e': skbio.diversity.alpha.heip_e,
+        'kempton_taylor_q': skbio.diversity.alpha.kempton_taylor_q,
+        'margalef': skbio.diversity.alpha.margalef,
+        'mcintosh_d': skbio.diversity.alpha.mcintosh_d,
+        'mcintosh_e': skbio.diversity.alpha.mcintosh_e,
+        'menhinick': skbio.diversity.alpha.menhinick,
+        'michaelis_menten_fit': skbio.diversity.alpha.michaelis_menten_fit,
+        'observed_otus': skbio.diversity.alpha.observed_otus,
+        'osd': skbio.diversity.alpha.osd,
+        'pielou_e': skbio.diversity.alpha.pielou_e,
+        'robbins': skbio.diversity.alpha.robbins,
+        'shannon': skbio.diversity.alpha.shannon,
+        'simpson': skbio.diversity.alpha.simpson,
+        'simpson_e': skbio.diversity.alpha.simpson_e,
+        'singles': skbio.diversity.alpha.singles,
+        'strong': skbio.diversity.alpha.strong,
+        'gini_index': skbio.diversity.alpha.gini_index,
+        'lladser_pe': skbio.diversity.alpha.lladser_pe,
+        'lladser_ci': skbio.diversity.alpha.lladser_ci}
+
+
+@experimental(as_of="0.4.0-dev")
+def get_alpha_diversity_metrics():
+    """ List scikit-bio's alpha diversity metrics
+
+    Returns
+    -------
+    list
+        Alphabetically sorted list of alpha diversity metrics implemented in
+        scikit-bio.
+
+    See Also
+    --------
+    get_beta_diversity_metrics
+
+    """
+    metrics = _get_alpha_diversity_metric_map()
+    return sorted(list(metrics.keys()))
+
+
+@experimental(as_of="0.4.0-dev")
+def get_beta_diversity_metrics():
+    """ List scikit-bio's beta diversity metrics
+
+    Returns
+    -------
+    list
+        Alphabetically sorted list of beta diversity metrics implemented in
+        scikit-bio.
+
+    See Also
+    --------
+    get_alpha_diversity_metrics
+    scipy.spatial.distance.pdist
+
+    Notes
+    -----
+    SciPy implements many additional beta diversity metrics that are not
+    included in this list. See documentation for
+    ``scipy.spatial.distance.pdist`` for more detail.
+
+    """
+    return ['unweighted_unifrac', 'weighted_unifrac']
+
+
+@experimental(as_of="0.4.0-dev")
+def alpha_diversity(metric, counts, ids=None, validate=True, **kwargs):
+    """ Compute alpha diversity for one or more samples
+
+    Parameters
+    ----------
+    metric: str, callable
+        The alpha diversity metric to apply to the sample(s).
+        Passing metric as a string is preferable as this often results in an
+        optimized version of the metric being used.
+    counts : 1D or 2D array_like of ints or floats
+        Vector or matrix containing count/abundance data. If a matrix, each row
+        should contain counts of OTUs in a given sample.
```

Comment:
At the last meeting we had in SD, we mostly agreed that these should be called *features* in the future, rather than *observations* for consistency with other fields/software. We decided to leave this as OTU for now, pending addition of the contingency table to scikit-bio. Prior to that, we'll have an RFC, and I just added [a note](https://github.com/biocore/scikit-bio-rfcs/issues/8) that we should discuss terminology in the RFC.